components/BookmarkButton.js

```diff
@@ -0,0 +1,51 @@
+import Image from "next/image";
+import styled, { css } from "styled-components";
+
+const StyledToggleBookmark = styled.button`
+  width: 40px;
+  height: 40px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-style: none;
+  border-radius: 0 0 5px 5px;
+`;
+
+const StyledToggleWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  right: 20px;
+  position: absolute;
+  z-index: 2;
+`;
```

Comment:
Please check if you really need the wrapper `<div>` or if you can position the `<button>` directly. 


**Little UI notes** ✏️

Although there’s no fixed size for a toggle (bookmark) button, there are some guidelines for interactive elements.

- For mobile, Google’s Material Design recommends a minimum touch [target size](https://m3.material.io/components/icon-buttons/specs#2b6575b0-cfa3-494e-9509-d7c1949d11f2) of 48x48 px
- For desktop, buttons can be smaller since mouse clicks allow for more precision. Typically, a 32-40 px height is a good baseline.
- The bookmark button should be large enough for easy access, especially if it’s a frequently used feature.